content/chapters/compute/lab/README.md

```diff
@@ -282,6 +282,201 @@ Therefore, there is also a gray area where there is no clear right answer to thi
+www-data      99       1     136  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data      99       1     137  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     101       1     101  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     101       1     138  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     101       1     139  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     101       1     140  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     101       1     141  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     101       1     142  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     101       1     143  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     105       1     105  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     105       1     126  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     105       1     127  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     105       1     130  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     105       1     131  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     105       1     132  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     105       1     133  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     109       1     109  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     109       1     115  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     109       1     116  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     109       1     121  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     109       1     122  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     109       1     123  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+www-data     109       1     124  0    7 21:07 pts/0    00:00:00 httpd -DFOREGROUND
+root         146       0     146  0    1 21:10 pts/1    00:00:00 bash
+root         152     146     152  0    1 21:10 pts/1    00:00:00 ps -efL
+```
+
+We see a much larger number of threads, as expected.
+
+### Practice: investigate apache2 using strace
+
+Use `strace` to discover the server document root. The document root is the path in the filesystem from where httpd serves all the files requested by the clients.
+
+First you will have to stop the running container using `make stop`, then restart it with `make run-privileged`.
+
+Then you will use `strace` inside the container to attach to the worker processes (use the `-p` option for this). You will also have to use `-f` flag with `strace`, so that it will follow all the threads inside the processes.
+
+After you have attached successfully to all worker processes, use the `curl` command to send a request, like the one in the beginning of this section.
+
+Then check the `strace` output to see what files were opened by the server.
+
```

Comment:
You can add a quiz here so students can check if they found the correct document root.